src/main/scala/au/com/agiledigital/toolform/tasks/generate/kubernetes/minikube/GenerateMinikube.scala

```diff
@@ -0,0 +1,60 @@
+package au.com.agiledigital.toolform.tasks.generate.kubernetes.minikube
+
+import java.io.{BufferedWriter, FileWriter}
+
+import au.com.agiledigital.toolform.app.{ToolFormConfiguration, ToolFormError}
+import au.com.agiledigital.toolform.model._
+import au.com.agiledigital.toolform.tasks.generate.kubernetes.minikube.DeploymentWriter.writeDeployment
+import au.com.agiledigital.toolform.tasks.generate.kubernetes.minikube.ServiceWriter.writeService
+import au.com.agiledigital.toolform.tasks.generate.{WriterContext, YamlWriter}
+import au.com.agiledigital.toolform.util.DateUtil
+import au.com.agiledigital.toolform.version.BuildInfo
+import cats.implicits._
+
+/**
+  * Takes an abstract project definition and outputs it a combined service/deployment Kubernetes spec in YAML format.
+  *
+  * For each service/resource defined in the Toolform config, a single Kubernetes service and deployment is created.
+  *
+  * Should be run via the GenerateTask class and not directly.
+  *
+  * @see https://kubernetes.io/docs/api-reference/v1.8/
+  */
+object GenerateMinikube extends YamlWriter {
+
+  private def shouldWriteService(toolFormService: ToolFormService): Boolean =
+    toolFormService.exposedPorts.nonEmpty || toolFormService.externalPorts.nonEmpty
+
+  /**
+    * The main entry point into the Kubernetes file generation.
+    *
+    * @param toolFormConfiguration  a configuration object that is parsed from command line options.
+    * @param project                the abstract project definition parsed by ToolFormApp.
+    * @return                       on success it returns a status message to print to the screen, otherwise it will return an
+    *                               error object describing what went wrong.
+    */
+  def runGenerateMinikube(toolFormConfiguration: ToolFormConfiguration, project: Project): Either[ToolFormError, String] = {
+    val outFile = toolFormConfiguration.generateTaskConfiguration.out
+    val sourceFilePath = toolFormConfiguration.in.getAbsolutePath
+    val writer = new BufferedWriter(new FileWriter(outFile, false))
+    try {
+      val writeFile = for {
+        _ <- write(s"# Generated by ${BuildInfo.name} (${BuildInfo.version})")
+        _ <- write(s"# Source file: $sourceFilePath")
+        _ <- write(s"# Date: ${DateUtil.formattedDateString}")
+        _ <- project.components.values.filter(shouldWriteService).toList.traverse_(writeService)
+        _ <- project.resources.values.filter(shouldWriteService).toList.traverse_(writeService)
+        _ <- project.components.values.toList.traverse_((component) => writeDeployment(project.id, component))
+        _ <- project.resources.values.toList.traverse_((resource) => writeDeployment(project.id, resource))
+      } yield ()
+
+      val context = WriterContext(writer)
+      // Final state needs to be read for anything to happen because of lazy evaluation
+      val _ = writeFile.run(context).value
+
+      Right("Completed Successfully")
```

Comment:
"Wrote configuration to [$outFile], run with ???" ?

Reply:
Fixed in 4b01e1c